pages/posts/website-analytics-snapshot-2023.md

```diff
@@ -0,0 +1,171 @@
+
+(For reasons I won't go into, we have a meta-schema for Hyper Schema in 2019-09, but not 2020-12.) 1.53 million to `.../meta/core`, and in the next place, the validation vocabulary meta-schema, with only 577.84k. If this is accurate, it's a little concerning. We would expect `/draft/2020-12/schema` to be the top path accessed, but it doesn't appear till position 8, with 472.46k.
+
+The traffic pattern is very spikey, with a repeating time based pattern. It's probably some cron job running. If we exclude the top IP address, we lose all the spikes, strongly suggesting this bursting traffic is coming from some Amazon controlled AWS servers.
+
+We'll reach out to AWS and see if we can get any more specific details. It's likely not a DoS attack, but it's a signal of something going wrong.
+
+## ASN, OSI, what?
+
+The top two ASN sources are Amazon controlled, totalling just over 39 million requests.
+
+For those of you who aren't network engineers, ASN is an [Autonomous System Number](https://www.cloudflare.com/en-gb/learning/network-layer/what-is-an-autonomous-system/), used for routing of packets over the internet, making sure they take the likely fastest route. If you cast your mind back to when you learnt about the OSI model (yeah, I don't remember much of that either)... [Open Systems Interconnection model](https://www.cloudflare.com/en-gb/learning/ddos/glossary/open-systems-interconnection-model-osi/), these ASNs are used as part of the transport layer.
+
+Point is, we can tell without needing to look up IP address owners, that almost 80% of our traffic is coming from AWS!
+
+## Country
+
+Given how much traffic is coming from AWS, it's not a surprise that about 43 million of those requests are coming from the US and Ireland. After those, Finland, India, and Netherlands make up the 1 mil club. Following on, Germany and Russia, and then several countries from Asia, all above 200k.
+
+## Source
+
+A staggering 51.12 million of those 51.92 million requests, are not declaring a reference. The following top three are json-schema.org, Google, and then, very interestingly, localhost!
+
+Weirdly, 1.54k requests declare a referrer of some defunct Uber settlement, just above Duck Duck Go.
+
+## What does this all mean?
+
+Access of meta-schemas is unlikely to corrolate with useage percentages, but it is at least an indicator of use to an extent.
+Looking at this data has meant we are reaching out to AWS to see if we can get any details.
+It has prompted us to reach out to IDE makers, although doing so was already on our list.
+We have extended our known platform users. It's nice when people use a proper agent string.
+
+Given we only have access to a 7 day history, there's not much we can tell over a longer time.
+While traditional analytics are useful, and we do use them, they can't tell us about accessing JSON files directly.
+
+While some of these numbers are interesting to see, I'm yet to be convinced that they provide much utility today.
```

Comment:
Considering that we don't have full clarity on the reasons of the traffic I'll suggest to end the article with a call to action to readers to provide feedback or ideas on the reasons of the data or maybe how to perform a deeper analysis. Do we have a github issue to continue the discussion there?

Reply:
Added. Let me know if you think this is OK by resolving this conversation and leaving an updated review? =]